cpp/foxglove/src/error.cpp

```diff
@@ -0,0 +1,51 @@
+#include <foxglove-c/foxglove-c.h>
+#include <foxglove/error.hpp>
+
+namespace foxglove {
+
+FoxgloveError::FoxgloveError(const foxglove_error&& error)
+    : std::exception()
+    , _impl(new foxglove_error(std::move(error))) {}
+
+FoxgloveError::~FoxgloveError() {
+  foxglove_error_free(_impl.get());
```

Comment:
this threw me off at first - but I now understand that the free function is freeing the message and has nothing to do with deleting the `new`'d memory from the constructor (the unique_ptr takes care of that). Could be worth some comments